ui/components/multichain/multi-srp/import-srp/import-srp.test.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import {
+  createEvent,
+  fireEvent,
+  RenderResult,
+  waitFor,
+} from '@testing-library/react';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { renderWithProvider } from '../../../../../test/jest/rendering';
+import mockState from '../../../../../test/data/mock-state.json';
+import * as actions from '../../../../store/actions';
+import { ImportSrp } from './import-srp';
+
+const mockClearClipboard = jest.fn();
+
+jest.mock('../../../../helpers/utils/util', () => ({
+  clearClipboard: () => mockClearClipboard(),
+}));
+
+const VALID_SECRET_RECOVERY_PHRASE =
+  'input turtle oil scorpion exile useless dry foster vessel knee area label';
+
+jest.mock('../../../../store/actions', () => ({
+  importMnemonicToVault: jest
+    .fn()
+    .mockReturnValue(jest.fn().mockResolvedValue(null)),
+  showAlert: jest.fn().mockReturnValue({ type: 'ALERT_OPEN' }),
+  hideAlert: jest.fn().mockReturnValue({ type: 'ALERT_CLOSE' }),
+}));
+
+const pasteSrpIntoFirstInput = (render: RenderResult, srp: string) => {
+  const [firstWord] = srp.split(' ');
+
+  const firstSRPWordDiv = render.getByTestId('import-srp__srp-word-0');
+  // This is safe because the input is always present in the word div.
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const firstSrpWordInput = firstSRPWordDiv.querySelector('input')!;
+
+  const pasteEvent = createEvent.paste(firstSrpWordInput, {
+    clipboardData: {
+      getData: () => srp,
+    },
+  });
+
+  fireEvent(firstSrpWordInput, pasteEvent);
+
+  return {
+    word: firstWord,
+    input: firstSrpWordInput,
+  };
+};
+
+describe('ImportSRP', () => {
+  const store = configureMockStore([thunk])(mockState);
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('enables the "Import wallet" button when a valid secret recovery phrase is entered', async () => {
+    const render = renderWithProvider(
+      <ImportSrp onActionComplete={jest.fn()} />,
+      store,
+    );
+    const { getByText } = render;
+
+    const importButton = getByText('Import wallet');
+
+    expect(importButton).not.toBeEnabled();
+    pasteSrpIntoFirstInput(render, VALID_SECRET_RECOVERY_PHRASE);
+    await waitFor(() => {
+      expect(importButton).toBeEnabled();
+    });
+  });
+
+  it('does not enable the "Import wallet" button when the secret recovery phrase is empty', async () => {
+    const render = renderWithProvider(
+      <ImportSrp onActionComplete={jest.fn()} />,
+      store,
+    );
+    const { getByText } = render;
+
+    expect(getByText('Import wallet')).not.toBeEnabled();
+    pasteSrpIntoFirstInput(render, '');
+    expect(getByText('Import wallet')).not.toBeEnabled();
```

Comment:
Nit:

As suggested here: https://github.com/MetaMask/metamask-extension/pull/30598/#discussion_r1975097915

We should use `importButton` in every tests?

```suggestion
    const importButton = getByText('Import wallet');

    expect(importButton).not.toBeEnabled();
    pasteSrpIntoFirstInput(render, '');
    expect(importButton).not.toBeEnabled();
```